src/Serilog.Sinks.File.Archive/ArchiveHooks.cs

```diff
@@ -70,12 +85,62 @@ public override void OnFileDeleting(string path)
                         sourceStream.CopyTo(compressStream);
                     }
                 }
+                //only apply archive file limit if we are archiving to a non tokenised path (constant path)
+                if (this.retainedFileCountLimit > 0 && !this.IsArchivePathTokenised)
+                {
+                    RemoveExcessFiles(currentTargetDir);
+                }
             }
             catch (Exception ex)
             {
                 SelfLog.WriteLine("Error while archiving file {0}: {1}", path, ex);
                 throw;
             }
         }
+
+        private bool IsArchivePathTokenised => this.targetDirectory is not null && TokenExpander.Expand(this.targetDirectory) != this.targetDirectory;
+
+        private void RemoveExcessFiles(string folder)
+        {
+            var searchPattern = this.compressionLevel != CompressionLevel.NoCompression
```

Comment:
Isn't there a risk of files being deleted that were not written by Serilog?

Reply:
yes, but without additional information from the file sink (log file pattern), or some special marker (like an attribute applied to the archived file) I don't see how this can be avoided across different process runs.

My plan was not to solve all possible issues up front, but to get a minimum viable extension to the hook that solves my immediate need.  i.e. compress files as they are rolled in the existing directory that is used just for the logs from one application.

Reply:
Thinking a bit more about this, if the archive hook is compressing the files it could put them into an actual zip file rather than just a gzipped stream and then we could add data in the "Extra" fields of the zip file to identify it later for removal.  

This could solve several issues at once; 
* the risk of removing files from other processes
* support for tokenised directory
* deterministic file ordering

I'll give it some more thought.

Reply:
Looks like the System.IO.Compression namespace does not expose the "extra data" fields.  A look at the source shows that it can roundtrip them but does not expose a way to read or write them so would need to go to a third party zip library to do this.  ICSharpCode.SharpZipLib can read and edit extra data for individual entries but I can't see that it exposes the central directory extra data which I think is more appropriate than the entry extra data, this is not a deal breaker as the file would have only 1 entry anyway.

Reply:
Hmm, on Windows there is also NTFS Alternate Data Streams, and on Linux there are extended attributes. Still, that makes assumptions about what is supported on end users' systems, and would be an extra layer of complexity.

I think if we just make clear in the docs that this feature is a footgun 🦶 🔫, then it's enough.